nengo/spa/actions.py

```diff
@@ -59,14 +56,21 @@ def validate_string(self, text):
     def __getitem__(self, key):
         # this gets used by the eval in the constructor to create new
         # terms as needed
-        item = self.objects.get(key, None)
-        if item is None:
-            if not key[0].isupper():
-                raise SpaParseError(
-                    "Semantic pointers must begin with a capital letter.")
-            item = Symbol(key)
-            self.objects[key] = item
-        return item
+        if key == '__tracebackhide__':  # gives better tracebacks in py.test
+            return False
+        if key in self.objects:
+            return self.objects[key]
+        else:
+            try:
```

Comment:
This logic took me a while to understand. An inline comment would be appreciated. Something like "attempt to create a submodule if it exists, otherwise create a normal object"?